drivers/platform/linux/linux_gpio.c

```diff
@@ -117,14 +120,26 @@ int32_t linux_gpio_get(struct gpio_desc **desc,
 	}
 
 	sprintf(path, "/sys/class/gpio/gpio%d/direction", descriptor->number);
-	linux_desc->direction_fd = open(path, O_WRONLY);
+	timeout = GPIO_TIMEOUT_MS;
+	while (--timeout) {
+		linux_desc->direction_fd = open(path, O_WRONLY);
+		if (linux_desc->direction_fd >= 0)
+			break;
+		mdelay(1);
+	}
 	if (linux_desc->direction_fd < 0) {
 		printf("%s: Can't open %s\n\r", __func__, path);
 		goto free_linux_desc;
 	}
 
 	sprintf(path, "/sys/class/gpio/gpio%d/value", descriptor->number);
-	linux_desc->value_fd = open(path, O_WRONLY);
+	timeout = GPIO_TIMEOUT_MS;
+	while (--timeout) {
+		linux_desc->value_fd = open(path, O_WRONLY);
+		if (linux_desc->value_fd >= 0)
+			break;
+		mdelay(1);
```

Comment:
is the delay required?

Reply:
I would say yes.
If I set a timeout without delay it will depend on the CPU speed.
I tested without delay I think 1000 times and it sometimes don't work.

Reply:
Then I believe a comment would be helpful pointing out why delay was added.
This approach I believe is used in other drivers also, if not specified in the datasheet.

Reply:
I edited the commit message